### PR TITLE
(WIP): Add a separate port for MTLS connection of pilot

### PR DESCRIPTION
**Pull request by wattli**

### CODE REVIEW

**Comment by ldemailly** [comment]
rebase/retest ?

**Comment by wattli** [comment]
It's still under implementation. Will ping back once it's ready for review. thanks.

File: install/kubernetes/helm/istio/charts/ingress/templates/deployment.yaml [hunk 1]
```diff
@@ -32,8 +32,10 @@ spec:
           - ingress
           - -v
           - "2"
-          - --discoveryAddress
+          - --discoveryPlainAddress
```
Comment (JAORMX) [inline install/kubernetes/helm/istio/charts/ingress/templates/deployment.yaml]: wouldn't it be better to keep the previous option (--discoveryAddress) and just add the --discoveryMtlsAddress option? that way folks could keep the previous templates.

Comment (costinm) [reply install/kubernetes/helm/istio/charts/ingress/templates/deployment.yaml]: +1  - no backward incompatible changes please.

File: install/kubernetes/helm/istio/charts/ingress/templates/deployment.yaml [hunk 2]
```diff
@@ -32,8 +32,10 @@ spec:
...
           - {{ .Release.Name }}-pilot:15003
+          - --discoveryMtlsAddress
```
Comment (costinm) [inline install/kubernetes/helm/istio/charts/ingress/templates/deployment.yaml]: I'm not sure I get this: why would we need both ? Based on policy we should select only one.

File: pilot/docker/envoy_pilot.json [hunk 3]
```diff
@@ -3,6 +3,27 @@
    {
     "address": "tcp://0.0.0.0:15003",
     "name": "tcp_0.0.0.0_15003",
+    "filters": [
```
Comment (costinm) [inline pilot/docker/envoy_pilot.json]: How about making this ( and the next one ) http listeners, so we get proper stats ?

**Pull request closed without merging**